src/test/java/com/rocketden/main/util/UtilityTestMethods.java

```diff
@@ -33,9 +37,15 @@ public static String convertObjectToJsonString(Object o) {
         Gson gson = new GsonBuilder().registerTypeAdapter(LocalDateTime.class,
             new JsonDeserializer<LocalDateTime>() { 
             @Override 
-            public LocalDateTime deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException { 
-                return LocalDateTime.parse(json.getAsString(),
-                    DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSS"));
+            public LocalDateTime deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+                    DateTimeFormatter formatter = new DateTimeFormatterBuilder()
+                        .appendPattern(DATE_TIME_FORMAT_PATTERN)
+                        // optional decimal point followed by 1 to 6 digits
+                        .optionalStart()
+                        .appendFraction(ChronoField.MICRO_OF_SECOND, 1, 6, true)
+                        .optionalEnd()
+                        .toFormatter(); 
+                    return LocalDateTime.parse(json.getAsString(), formatter);
```

Comment:
Does this fix those constant test failiures? If so that's awesome - how does this work?

Reply:
Yes I think it does (I should've figured this out earlier tbh), here's how it works:

The Date used to be parsed by the following format: `yyyy-MM-dd'T'HH:mm:ss.SSSSSS". It is notable that there are exactly six microseconds there -- if there are any less, than the test fails because it cannot parse it. So I believe what was happening (I could be wrong, but something like this) was occasionally the last `S` would be a `0`, which means the format was only five `S`, and thus the test failed.

This setup correctly parses correctly regardless of how many microseconds there are, so it should parse as expected now.